pkg/controller/deployment/deployment_controller.go

```diff
@@ -491,6 +498,11 @@ func (dc *DeploymentController) syncRollingUpdateDeployment(deployment extension
 		return dc.updateDeploymentStatus(allRCs, newRC, deployment)
 	}
 
+	if deployment.Spec.RevisionHistoryLimit != nil {
+		// Cleanup old RCs
+		return dc.cleanupOldRcs(oldRCs, deployment)
```

Comment:
Actually now that I am looking at this carefully, deployments with cleanup policy will never reach `syncDeploymentStatus` right? I think we shouldn't return here in any case; either if there is no error returned from `cleanupOldRcs` or even in the case of an error we should just log it and proceed to update the status. Thoughts?